subsys/logging/Kconfig.misc

```diff
@@ -94,6 +94,15 @@ config LOG_DOMAIN_NAME
 	string "Domain name"
 	default ""
 
+config LOG_CUSTOM_HEADER
+	bool "Include Custom Log Header"
+	help
+		If enabled, a custom application defined header, named "zephyr_custom_log.h",
+		will be included at the end of log.h. This allows for the LOG_* macros to be
+		redefined at the macro level. The application must provide a header named
+		"zephyr_custom_log.h" when this option is enabled. The internal LOG API may
+		change, so use with caution.
```

Comment:
I would change the last sentence to something like:

The LOG frontend API can be modified by calls to undef and then remapping the entry points to new functionality. This should be done cautiously.

Reply:
Reworded a bit and added a paragraph to logging/index.rst documentation.